### DISCUSSION TITLE
picotls support

### DISCUSSION THREAD

**Discussion by amiran-malania** [ngtcp2/ngtcp2#1308] [Q&A]
Hi!

Is picotls still supported as a TLS backend? I keep getting "Disabling Picotls because ptls_openssl_random_bytes not found in picotls/build/libpicotls-core.a  picotls/build/libpicotls-openssl.a". I'm switching to correct branch of picotls before building it, as stated in README.

Can anybody provide correct guide to for the build?

**Comment by tatsuhiro-t** [comment]
It should work.  Our CI builds client and server with picotls backend successfully.

**Reply by amiran-malania** [reply]
I took a look at CI scripts and adjusted it to my needs, I have it working for Apple M1 but the it fails with the same error for Linux.

**Reply by tatsuhiro-t** [reply]
I got received similar reports several times, but it turned out almost always there are something wrong in their environments like environment variable pointing at the wrong direction, etc.

**Reply by amiran-malania** [reply]
It's mentioned in README that you use picotls with OpenSSL as crypto backend but it's not documented(in readme) how to correctly build the library against picotls. Is it possible to provide the steps for it?

**Reply by tatsuhiro-t** [reply]
Check https://github.com/ngtcp2/ngtcp2/blob/main/.github/workflows/build.yml to see how to bulid picotls in our CI builds.
If you have a linking issue against openssl, it most likely you forget to add a path to the directly where openssl.pc is installed to PKG_CONFIG_PATH.
